config/scheme_test.go

```diff
@@ -0,0 +1,31 @@
+package config
+
+import (
+	"testing"
+
+	"golang.ngrok.com/ngrok/internal/tunnel/proto"
+)
+
+func TestScheme(t *testing.T) {
+	cases := []testCase[httpOptions, proto.HTTPEndpoint]{
+		{
+			name:        "default",
+			opts:        HTTPEndpoint(),
+			expectProto: stringPtr(string(SchemeHTTPS)),
+		},
+		{
+			name:        "set https",
+			opts:        HTTPEndpoint(WithScheme(SchemeHTTPS)),
+			expectProto: stringPtr(string(SchemeHTTPS)),
+		},
+		{
+			name:        "force http",
+			opts:        HTTPEndpoint(WithScheme(SchemeHTTP)),
+			expectProto: stringPtr(string(SchemeHTTP)),
+		},
```

Comment:
should probably add "with both"

Reply:
That's not a thing in the agent libraries - users just bind both tunnels individually.